### DISCUSSION TITLE
GitHub OAuth access token request returns errors with status code 200

### DISCUSSION THREAD

**Discussion by cmd-johnson** [community/community#57068] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When trying to exchange an invalid `code` for an access token, GitHub's token endpoint (https://github.com/login/oauth/access_token) correctly returns an error body indicating that the code was invalid. However, the status code of the response is set to `200`.

Since the request returned an error, I would've expected a `4xx` status code instead, as specified in the [OAuth2 spec](https://www.rfc-editor.org/rfc/rfc6749#section-5.2):

```sh
curl -i --data 'client_id=[omitted]&client_secret=[omitted]&redirect_uri=[omitted]&code=invalidCode' https://github.com/login/oauth/access_token
HTTP/2 200 
server: GitHub.com
content-type: application/x-www-form-urlencoded; charset=utf-8
[other headers omitted]

error=bad_verification_code&error_description=The+code+passed+is+incorrect+or+expired.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-oauth-app-access-token-request-errors%2F%23bad-verification-code
```

**Comment by jdtheefirst** [comment]
if you have binary (non-alphanumeric) data (or a significantly sized payload) to transmit, use `multipart/form-data`. Otherwise, use `application/x-www-form-urlencoded`.

**Reply by HoaX7** [reply]
im using application/json
but it still weird why the misleading status code is returned

**Comment by tilda-deimos** [comment]
I have the same issue. 200 status code with an error JSON response.

**Reply by HoaX7** [reply]
same issue here, in golang, wtf?

**Comment by HoaX7** [comment]
is there any update on this? Im facing the same issue. It looks like its been too long

**Comment by burabatbold** [comment]
Any solution guys , i'm facing this issue.

**Comment by LorenzoLeonardo** [comment]
Sounds like github didn't follow the RFC specs

**Comment by caner-cetin** [comment]
unanswered by the incompetent slop generators from microsoft, thanks guys

why not just fuck all RFC specs, return 200 on every single GitHub API request, not just OAuth, and return a `{success:"false"}` field, make it string for extra robustness and scalability. for fucking sake.
